app/src/main/java/com/mybus/service/MyBusService.java

```diff
@@ -0,0 +1,46 @@
+package com.mybus.service;
+
+import com.google.android.gms.maps.model.LatLng;
+import com.mybus.builder.MyBusServiceUrlBuilder;
+
+import org.json.JSONException;
+import org.json.JSONObject;
+
+import java.io.IOException;
+
+public class MyBusService extends GenericService {
+
+    /**
+     * @param origin
+     * @param destiny
+     * @return
+     */
+    public JSONObject searchRoutes(LatLng origin, LatLng destiny) throws IOException, JSONException {
```

Comment:
The signature should be:  

```java
public List<BusRouteResult> searchRoutes(LatLng origin, LatLng destination)...
```

The parsing from JSON to model classes should be resolved inside the service.